pkg/segment/reader/record/recordreader.go

```diff
@@ -166,7 +170,12 @@ func GetRecordsFromSegment(segKey string, vTable string, blkRecIndexes map[uint1
 
 		for r := range resultAllRawRecs {
 			resultAllRawRecs[r][config.GetTimeStampKey()] = recordIdxTSMap[r]
-			resultAllRawRecs[r]["_index"] = vTable
+
+			// if the total column count is less than or equal to the number of columns we have read, it indicates there is no field operations.
+			if totalColumnCount <= len(allMatchedColumns) {
+				resultAllRawRecs[r]["_index"] = vTable
+				addedExtraFields = true
+			}
```

Comment:
The truth of this comment isn't obvious to me. Can you explain why this is true?

Reply:
**totalColumnCount** has count of all the columns before applying the filter. The condition checks whether the **allMatchedColumns** has the same number or more columns. If its **allMatchedColumns** has less number of columns than **totalColumnCount**, we can safely say **fields** query was used.

Reply:
I understand what the code and comment say, I just don't understand why it's correct. _Why_ can we infer `fields` was not used if `totalColumnCount <= len(allMatchedColumns)`?

What if `fields` was used to select every single column, or if `eval` (see https://docs.splunk.com/Documentation/Splunk/9.1.1/SearchReference/Eval) was used to create multiple columns, and then `fields` was used to select any number of columns (either more or fewer columns that the number of raw columns in the segment)?